package.json

```diff
@@ -205,6 +177,21 @@
                 "when": "editorTextFocus && inputFocus && notebookEditorFocused && notebookType == jupyter-notebook && config.jupyter.enableKeyboardShortcuts",
                 "command": "notebook.cell.execute"
             },
+            {
+                "key": "ctrl+enter",
+                "when": "inputFocus && notebookEditorFocused && !editorHasMultipleSelections && !editorHasSelection && !editorHoverVisible && notebookCellType == 'markup'",
+                "command": "notebook.cell.quitEdit"
+            },
+            {
+                "key": "shift+enter",
+                "when": "editorTextFocus && inputFocus && notebookEditorFocused && notebookType == jupyter-notebook && config.jupyter.enableKeyboardShortcuts || notebookCellListFocused && notebookCellType == 'markup' || notebookCellListFocused && notebookCellType == 'code' && notebookCellExecutionState == 'failed' || notebookCellExecutionState == 'idle' || notebookCellExecutionState == 'succeeded' && notebookKernelCount > 0",
```

Comment:
This when statement strikes me as a bit gnarly to parse. But not sure if we can to anything to change that.

Reply:
Yeah, that's a merge of the when clauses for our existing ctrl+enter keybinding and VS Code's shift+enter keybinding.